lib/rubygems/ext/builder.rb

```diff
@@ -56,23 +56,23 @@ def self.make(dest_path, results)
   end
 
   def self.redirector
+    warn "#{caller[0]}: Use IO.popen(..., err: [:child, :out])"
     '2>&1'
   end
 
   def self.run(command, results, command_name = nil)
     verbose = Gem.configuration.really_verbose
 
     begin
-      # TODO use Process.spawn when ruby 1.8 support is dropped.
       rubygems_gemdeps, ENV['RUBYGEMS_GEMDEPS'] = ENV['RUBYGEMS_GEMDEPS'], nil
       if verbose
         puts("current directory: #{Dir.pwd}")
         puts(command)
         system(command)
       else
         results << "current directory: #{Dir.pwd}"
-        results << command
-        results << `#{command} #{redirector}`
+        results << (Array === command ? command.shelljoin : command)
```

Comment:
Is there any reason to prefer `Array === command` over `command.is_a?(Array)`? If not, could you switch to the latter?